pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -42,6 +43,22 @@ func validName(ctx Context) validation.Result {
 	return validation.OK
 }
 
+func specUpdatedToBeta(ctx Context) validation.Result {
+	oldAPIVersion := "elasticsearch.k8s.elastic.co/v1alpha1"
+
+	es := ctx.Proposed.Elasticsearch
+	if es.APIVersion == oldAPIVersion || es.Spec.NodeSets == nil {
+		return validation.Result{Reason: validationFailedMsg}
```

Comment:
Since we know exactly what the reason is, I think we should include it in the message here